core/src/main/java/org/bitcoinj/core/PrefixedChecksummedBytes.java

```diff
@@ -82,8 +82,10 @@ public PrefixedChecksummedBytes clone() throws CloneNotSupportedException {
 
     /**
      * This implementation uses an optimized Google Guava method to compare {@code bytes}.
+     * @deprecated Can't implement comparable because of Segwit changes (subclasses with
+     * added value components, eg. version field based on p2sh in LegacyAddress)
      */
-    @Override
+    @Deprecated
     public int compareTo(PrefixedChecksummedBytes o) {
```

Comment:
Do we still need this old impl? Since the `Comparable` interface has been removed from this class, I assume it would not serve backwards compatibility much?

Reply:
It's not needed internally. I don't know if there are any external projects depending on it. I had no choice but to remove the `Comparable` interface (because I couldn't add it to `Address` otherwise) but I left this here just in case some other project needs the comparator code. But I like the idea of deleting it.

Reply:
Fixed in 2a204b79221be0292a7fe1ce3c8ed97c045b1501 -- I'll squash when everything else is resolved and we change the PR from "Draft" to "Ready to Merge".